src/VisualStudio/Core/Def/LanguageService/AbstractPackage`2.cs

```diff
@@ -66,9 +69,15 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             return _languageService.ComAggregate!;
         });
 
+        var miscellaneousFilesWorkspace = this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
+
         await shell.LoadPackageAsync(Guids.RoslynPackageId);
 
-        var miscellaneousFilesWorkspace = this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
+        // Be a bit paranoid, as LoadPackageAsync returns an IVsTask, and I'm not certain of it's
+        // behavior if call finishes on a different thread. This should be removed soon anyway as the code
+        // following this looks like it can be reordered/refactored to not require the main thread.
```

Comment:
might want to check in with Ryan/Dave/Andrew about expectations.  given a naked `await` (and this being IVS stuff) i would always expect CA(true) semantics.